packages/compass-indexes/src/modules/create-index.tsx

```diff
@@ -705,11 +776,17 @@ const reducer: Reducer<State, Action> = (state = INITIAL_STATE, action) => {
   if (
     isAction<CreateIndexOpenedAction>(action, ActionTypes.CreateIndexOpened)
   ) {
+    const parsedInitialQuery = action.initialQuery
+      ? JSON.stringify(action.initialQuery, null, 2)
+      : '';
+
     return {
       ...getInitialState(),
       isVisible: true,
-      query: action.query ?? null,
-      currentTab: action.query ? 'QueryFlow' : 'IndexFlow',
+      // get it from the current query or initial query from insights nudge
+      query: state.query || parsedInitialQuery,
```

Comment:
Not sure, but shouldn't `parsedInitialQuery` take precedence here? Like if something triggered opening a modal with a new query, it should replace whatever we had in the state before?